app/src/main/java/com/github/swent/echo/compose/components/searchmenu/SearchMenuFilters.kt

```diff
@@ -0,0 +1,198 @@
+        }
+    }
+}
+
+/** Data class for the checkboxes */
+data class CheckBoxItems(
+    val icon: ImageVector,
+    val contentDescription: String,
+    var checked: MutableState<Boolean>
+)
+
+/** Composable to display the checkboxes in the correct format */
+@Composable
+fun CheckBoxesDisplayer(title: String, checkBoxItems: List<CheckBoxItems>) {
+    Column {
+        Text(title, modifier = Modifier.testTag("checkboxes_title"))
+        Spacer(modifier = Modifier.height(10.dp))
+        checkBoxItems.forEach { checkBoxItem ->
+            Row(modifier = Modifier.testTag("${checkBoxItem.contentDescription}_checkbox_row")) {
+                Icon(checkBoxItem.icon, contentDescription = checkBoxItem.contentDescription)
+                Checkbox(
+                    checked = checkBoxItem.checked.value,
+                    onCheckedChange = { checkBoxItem.checked.value = !checkBoxItem.checked.value },
+                    modifier =
+                        Modifier.height(25.dp)
+                            .width(25.dp)
+                            .testTag("${checkBoxItem.contentDescription}_checkbox")
+                )
+                Text(checkBoxItem.contentDescription)
+            }
+            Spacer(modifier = Modifier.height(5.dp))
+        }
+    }
+}
+
+/** Composable to display the dropdown menu for the sort by filter */
+@Composable
+fun SortByDisplayer(sortBy: MutableState<SortBy>) {
+    var expanded by remember { mutableStateOf(false) }
+
+    Column {
+        Button(
+            onClick = { expanded = !expanded },
+            shape = RoundedCornerShape(10),
+            modifier = Modifier.width(170.dp).testTag("sort_by_button")
+        ) {
+            Text(
+                if (sortBy.value == SortBy.NONE)
+                    stringResource(id = R.string.search_menu_filters_sort_by)
+                else stringResource(id = stringResourceSortBy(sortBy.value.stringKey))
+            )
+            Icon(
+                if (!expanded) Icons.Filled.KeyboardArrowDown else Icons.Filled.KeyboardArrowUp,
+                contentDescription = stringResource(id = R.string.search_menu_filters_sort_by)
+            )
+        }
+        // Check if the sort by filter is expanded
+        if (expanded) {
+            SortBy.entries.forEach {
+                Button(
+                    onClick = {
+                        sortBy.value = it
+                        expanded = false
+                    },
+                    shape = RoundedCornerShape(5),
+                    modifier = Modifier.width(170.dp).height(35.dp).testTag(it.stringKey)
+                ) {
+                    Text(it.stringKey)
+                }
+            }
+        }
+    }
+}
+
+/*
+ * Composable to get a date from the user.
+ * TODO : Implement it, but I spend a bit too much time
+ *   on non-working solution so leaving it for next sprint
+ */
+@Composable fun DateInputSample(dateOutput: MutableState<ZonedDateTime>) {}
```

Comment:
Hi, for the time selector slider, this component looks like it does what we need: https://developer.android.com/develop/ui/compose/components/slider
Did you try this?